plugin.py

```diff
@@ -46,5 +47,49 @@ def on_pre_server_command(self, params: Mapping[str, Any], done_callback: Callab
             done_callback()
         return True
 
+    def on_server_response_async(self, method: str, response: Response) -> None:
+        if method == 'codeAction/resolve':
+            self.maybe_handle_move_code_action(cast(CodeAction, response.result))
 
+    def maybe_handle_move_code_action(self, code_action: CodeAction) -> None:
+        session = self.weaksession()
+        if not session:
+            return
+        command = cast(MoveFunctionCommand, code_action.get('command'))
+        if not command:
+            return
+        command_name, params, function_name = command.get('arguments')
+        if command_name == 'moveFunction':
```

Comment:
I'm not sure if you missed my comment but I was saying that this method is called when any code action is resolved so we don't know at this point whether `arguments` is gonna be a list with 3 items. I think the code should be more defensive.

Based on the [code you've linked](https://github.com/elm-tooling/elm-language-server/blob/171c1399a0327e4381a2052e268037ec5459ca37/src/providers/codeAction/moveFunctionCodeAction.ts#L40-L43) you could check that code action's title is `Move Function` and only then proceed to extract arguments.

Reply:
Missed that, thanks 👍
will push a commit later